File: Python3/프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
import math

def solution(left, right):
    return sum(i if isEven(i) else -i for i in range(left, right + 1))

def isEven(num: int) -> bool:
    nset = set()
    for i in range(1, int(math.sqrt(num) + 1)):
        if num % i == 0:
            nset.add(i)
            nset.add(int(num / i))
    # print(num,"의 약수는", nset)
    return len(nset) % 2 == 0

File: C++/프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.cpp
#include <set>
#include <cmath>

using namespace std;

bool isEven(int num) {
    set<int> submultiple;
    
    for(int i = 1; i <= sqrt(num); i++) {
        if(num % i == 0) {
            submultiple.emplace(i);
            submultiple.emplace(num / i);
        }
    }
    
    return submultiple.size() % 2 == 0;
}

int solution(int left, int right) {
    int answer = 0;
    
    for(int i = left; i <= right; i++) {
        answer += isEven(i) ? i : -i;
    }
    return answer;
}
